htlcswitch/iterator.go

```diff
@@ -155,11 +186,23 @@ func NewOnionProcessor(router *sphinx.Router) *OnionProcessor {
 	return &OnionProcessor{router}
 }
 
+// Start spins up the sphinx router which for processing onion packets and
+// detecting duplicate key reuse.
+func (p *OnionProcessor) Start() error {
```

Comment:
I don't think this is the proper place for this method. We already instantiate the `OnionProcessor` within the `server` struct. So within the server's `Start` and `Stop` method, we can take care of this.

Reply:
Good call, fixed